test/integration/scenarios/10.namespace/magefile.go

```diff
@@ -0,0 +1,125 @@
+//go:build mage
+
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/eclipse-symphony/symphony/test/integration/lib/testhelpers"
+	"github.com/princjef/mageutil/shellcmd"
+)
+
+// Test config
+const (
+	TEST_NAME    = "Symphony catalog namespace test scenario"
+	TEST_TIMEOUT = "4m"
+)
+
+var (
+	NAMESPACES = []string{
+		"default",
+		"nondefault",
+	}
+)
+
+var (
+	// catalogs to deploy
+	testManifests = []string{
+		"manifest/config-container.yaml",
+		"manifest/config.yaml",
+		"manifest/campaign-container.yaml",
+		"manifest/campaign.yaml",
+	}
+
+	testActivation = "manifest/activation.yaml"
+
+	// Tests to run
+	testVerify = []string{
+		"./verify/...",
+	}
+)
+
+// Entry point for running the tests
+func Test() error {
+	fmt.Println("Running ", TEST_NAME)
+
+	defer testhelpers.Cleanup(TEST_NAME)
+
+	err := testhelpers.SetupCluster()
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range NAMESPACES {
+		os.Setenv("NAMESPACE", namespace)
+		err := DeployManifests(namespace)
+		if err != nil {
+			return err
+		}
+		err = Verify()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Verify namespace in nondefault namespace
+func DeployManifests(namespace string) error {
+	// Ensure that namespace is defined
+	err := testhelpers.EnsureNamespace(namespace)
+	if err != nil {
+		return err
+	}
+	os.Setenv("NAMESPACE", namespace)
+
+	// setup campaign
+	currentPath, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	for _, manifest := range testManifests {
+		absManifest := filepath.Join(currentPath, manifest)
+		err := shellcmd.Command(fmt.Sprintf("kubectl apply -f %s -n %s", absManifest, namespace)).Run()
+		if err != nil {
+			return err
+		}
+	}
+
+	// wait for 5 seconds to make sure campaign is created
```

Comment:
Why wait instead of polling status？

Reply:
removed. The campaign only has mock stage and has no need to wait for status here.